tensorflow/core/kernels/scatter_nd_op.cc

```diff
@@ -360,31 +381,6 @@ class IndexFlattener {
   }
 };
 
-#ifdef TENSORFLOW_USE_SYCL
-template <typename Index>
-class IndexFlattener<SYCLDevice, Index> {
```

Comment:
I remember I introduced this a long time when we didn't know HostMemory("indices") was a thing so it's probably better this way.
You you also remove the whole class IndexFlattener and just inline the function call in DoScatterNd.
But were you able to test if this works? scatter_nd_ops_test is a test that fails for me on dev/amd_gpu for other reasons it seems. John reported it in issue TF-143.

Reply:
Did we introduce IndexFlattener?

Reply:
Yes

Reply:
Only the partial specialisation tho.. I did reverted it already.

Reply:
I'm 99% sure I introduced it but now we can't see it because of all the merges that has been going on.
It wouldn't make sense for non-SYCL people to have this class.

Reply:
I am referring to this class: https://github.com/lukeiwanski/tensorflow/blob/043a2c8fb2be00e562d2da2c25bd75140556ada7/tensorflow/core/kernels/scatter_nd_op.cc#L376

I did remove our partial specialisation that was doing memcopies. Now TF handles that and goes through original class.

Reply:
Yes I am referring to this class as well. There are no other usages than right below in DoScatterNd. I introduced it because I didn't want to copy the whole DoScatterNd function for such a small change.

Reply:
Ok interesting: https://github.com/lukeiwanski/tensorflow/commit/f5ceb90e7f08fbe7605a002a546b22ef893f248c I will leave it there for now..